kdbx/src/main/java/org/linguafranca/pwdb/kdbx/KdbxKeyFile.java

```diff
@@ -45,33 +52,61 @@ public class KdbxKeyFile {
      * @return the key
      */
     public static byte[] load(InputStream inputStream) {
+
         try {
-            DocumentBuilder documentBuilder = DocumentBuilderFactory.newInstance().newDocumentBuilder();
-            Document doc = documentBuilder.parse(inputStream);
-            String version = (String) xpath.evaluate("//KeyFile/Meta/Version/text()", doc, XPathConstants.STRING);
-            String data = (String) xpath.evaluate("//KeyFile/Key/Data/text()", doc, XPathConstants.STRING);
-            if (data == null) {
-                return null;
-            }
-            if (version.equals("2.0")) {
-               
-                byte[] hexData = Hex.decodeHex(data.replaceAll("\\s",""));
-               
-                MessageDigest md = Encryption.getSha256MessageDigestInstance();
-                byte[] computedHash = md.digest(hexData);
-               
-                String hashToCheck = (String) xpath.evaluate("//KeyFile/Key/Data/@Hash", doc, XPathConstants.STRING);
-                byte[] verifiedHash = Hex.decodeHex(hashToCheck);
-                
-                for(int i = 0; i < verifiedHash.length; i++) {
-                    if(computedHash[i] != verifiedHash[i]) {
+            byte[] inputBytes = ByteStreams.toByteArray(inputStream);
+            if (inputBytes.length == 32) {
+                //32 bytes KeyFile
+                return inputBytes;
+            } else if (inputBytes.length == 64) {
+                //64 bytes KeyFile (only Hexadecimal values)
+                if (Helpers.isValidHexString(inputBytes)) {
+                    byte[] keyFile = org.bouncycastle.util.encoders.Hex.decode(inputBytes);
+                    return keyFile;
+                } else {
+                    throw new IllegalStateException("KeyFile contains not allowed characters");
```

Comment:
presumably, bouncy castle throws an exception if non-valid hex, so the additional test is at least redundant, and may in fact reject hex with whitespace that bouncy castle would accept e.g. "ab cd ef 12 34 ..."

Reply:
See comment above